scikeras/utils/transformers.py

```diff
@@ -390,3 +407,31 @@ def get_metadata(self):
             "n_outputs_": self.n_outputs_,
             "n_outputs_expected_": self.n_outputs_expected_,
         }
+
+
+class ClassWeightDataTransformer(BaseEstimator, TransformerMixin):
+    """Default dataset_transformer for KerasClassifier.
+
+    This transformer implements handling of the `class_weight` parameter
+    for single output classifiers.
+    """
+
+    def __init__(self, class_weight: Optional[Union[str, Dict[int, float]]] = None):
+        self.class_weight = class_weight
+
+    def fit(
+        self,
+        data: Tuple[np.ndarray, Optional[np.ndarray], Optional[np.ndarray]],
+        dummy: None = None,
+    ) -> "ClassWeightDataTransformer":
+        return self
+
+    def transform(
+        self, data: Tuple[np.ndarray, Optional[np.ndarray], Optional[np.ndarray]]
+    ) -> Tuple[np.ndarray, Union[np.ndarray, None], Union[np.ndarray, None]]:
+        X, y, sample_weight = data
+        if self.class_weight is None or y is None:
+            return (X, y, sample_weight)
+        sample_weight = 1 if sample_weight is None else sample_weight
+        sample_weight *= compute_sample_weight(class_weight=self.class_weight, y=y)
+        return (X, y, sample_weight)
```

Comment:
There is another option here, that may be cleaner, which would be to add another dependency injection point that runs in "parallel" to `feature_encoder` and `target_encoder` specifically for `sample_weight`. But since it also makes sense that one would need at least `y` to process `sample_weight`, it might be a bit redundant.